### PR TITLE
ConvertPadToGroupConvolution support Pad12 with positive indexes

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 - fix ConvertPadToGroupConvolution to support Pad12 with positive indexes
 - fix unit tests to check both v1::Pad and v12::Pad
 - add unit test to check negative padding case

### Tickets:
 - 112279

### CODE REVIEW

File: src/common/transformations/src/transformations/op_conversions/convert_pad_to_group_conv.cpp [hunk 1]
```diff
@@ -57,6 +58,15 @@ ov::pass::ConvertPadToGroupConvolution::ConvertPadToGroupConvolution() {
             return false;
         }
 
+        // check that Pad has non-negative values
```
Comment (itikhono) [inline src/common/transformations/src/transformations/op_conversions/convert_pad_to_group_conv.cpp:61]: Are these checks absolutely the same as in AvgPool, Convolution Fusion transformations?
Maybe it's better to move them in one place

**Review by itikhono — APPROVED** [review]

**Review by t-jankowski — APPROVED** [review]

File: src/common/transformations/tests/op_conversions/convert_pad_to_group_conv.cpp [hunk 2]
```diff
@@ -5,104 +5,227 @@
...
+std::vector<PadFactoryPtr> pad_factories = {CREATE_PAD_FACTORY(ov::op::v1::Pad, "op_v1_Pad"),
+                                            CREATE_PAD_FACTORY(ov::op::v12::Pad, "op_v12_Pad")};
+
+struct ITestModelFactory {
+    explicit ITestModelFactory(const std::string& a_test_name) : test_name(a_test_name) {}
+    virtual ~ITestModelFactory() = default;
+    virtual void setup(PadFactoryPtr pad_factory, ov::pass::Manager& manager) = 0;
+    std::string test_name;
+    std::shared_ptr<ov::Model> function;
+    std::shared_ptr<ov::Model> function_ref;
```
[the remark this answers is not in the document]
Comment (itikhono) [reply src/common/transformations/tests/op_conversions/convert_pad_to_group_conv.cpp:85]: I think it's better to do it in the next PR
